### PR TITLE
Task/task 71-implement two factor authentication has been created

### PR DESCRIPTION
**Pull request by jananikuruntham**

### CODE REVIEW

File: api/src/mail/mail.service.ts [hunk 1]
```diff
@@ -46,4 +46,30 @@ export class MailService {
...
+        subject: 'One Time Password (OTP) - AUTHMIKA',
+        template: './otp-email-template',
+        context: {
+          userName: user_name,
+          displayName: display_name,
+          otp: otp,
+          url: url,
+        },
+      });
+      console.log('email', otp);
```
Comment (vsaravanakumar) [inline api/src/mail/mail.service.ts:69]: Remove the unwanted console.log

File: api/src/users/users.service.ts [hunk 2]
```diff
@@ -8,13 +8,16 @@ import {
...
 import { InjectModel } from '@nestjs/sequelize';
-import { Optional } from 'sequelize';
+import { Op, Optional } from 'sequelize';
 import { Users } from 'src/db/model/users.model';
 import { UsersDto } from './dto/users.dto';
 import { hash } from 'bcrypt';
 import { UserApplications } from 'src/db/model/user-applications.model';
 import { compare } from 'bcrypt';
 import { Applications } from 'src/db/model/applications.model';
+import * as speakeasy from 'speakeasy';
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:18]: Add package.json file to install this library

File: api/src/users/users.service.ts [hunk 3]
```diff
@@ -8,13 +8,16 @@ import {
...
+import { randomBytes } from 'crypto';
+// import { MailerService } from '@nestjs-modules/mailer';
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:20]: Remove the commented line if not need

File: api/src/users/users.service.ts [hunk 4]
```diff
@@ -25,8 +28,80 @@ export class UsersService {
     private userApplictionsModel: typeof UserApplications,
     @InjectModel(Applications)
     private readonly applictionsModel: typeof Applications,
+    // private mailerService: MailerService,
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:31]: Remove the commented line if not need

File: api/src/users/users.service.ts [hunk 5]
```diff
@@ -99,20 +175,29 @@ export class UsersService {
...
           mobile: mobile,
           role: role,
-          createdBy: user ? user.id : null,
+          createdBy: null,
         });
 
         if (!newUser) {
           throw new InternalServerErrorException('User creation failed');
+        } else {
+          // const url = `${process.env.BASE_URL}/login`;
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:184]: Remove the commented line if not need

File: api/src/users/users.service.ts [hunk 6]
```diff
@@ -131,6 +216,122 @@ export class UsersService {
     }
   }
 
+  // async sendPasswordByEmail(
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:219]: Remove the commented line if not need

File: api/src/users/users.service.ts [hunk 7]
```diff
@@ -131,6 +216,122 @@ export class UsersService {
...
+  //         url: url,
+  //       },
+  //     });
+  //     return true;
+  //   } catch {
+  //     return false;
+  //   }
+  // }
+
+  // async sendOtp(
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:244]: Remove the commented line if not need

File: api/src/users/users.service.ts [hunk 8]
```diff
@@ -131,6 +216,122 @@ export class UsersService {
...
+  //       return {
+  //         message: error.message,
+  //         statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+  //         data: {},
+  //       };
+  //     }
+  //   }
+  // }
+
+  // async sendOtpByEmail(
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:298]: Remove the commented line if not need

File: api/src/mail/mail.service.ts (hunk 1 above)
Comment (jananikuruntham) [reply api/src/mail/mail.service.ts]: removed console.log

File: api/src/users/users.service.ts (hunk 2 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts:18]: added package.json file

File: api/src/users/users.service.ts (hunk 3 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed commented lines

File: api/src/users/users.service.ts (hunk 4 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed unwanted commented lines

File: api/src/users/users.service.ts (hunk 5 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed commented lines

File: api/src/users/users.service.ts (hunk 6 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed unwanted commented lines

File: api/src/users/users.service.ts (hunk 7 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed unwanted commented lines

File: api/src/users/users.service.ts (hunk 8 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed unwanted commented lines

**Pull request merged**